imod/mf6/wel.py

```diff
@@ -494,18 +492,32 @@ def regrid_like(self, target_grid: GridDataArray, *_) -> "Well":
         grid-agnostic, instead this method clips the package based on the grid
         exterior.
         """
-        return clip_by_grid(self, target_grid)
+        target_grid_2d = target_grid.isel(layer=0, drop=True, missing_dims="ignore")
+        return clip_by_grid(self, target_grid_2d)
 
-    def mask(self, _) -> Package:
+    def mask(self, domain: GridDataArray) -> "Well":
```

Comment:
If you add `from __future__ import annotations` then you can add the Well type hint as a regular hint instead of a string: `-> Well`